src/state/actions/dispatchArIOInteraction.ts

```diff
@@ -202,60 +187,64 @@ export default async function dispatchArIOInteraction({
-          payload: 'Confirming Primary Name 1/2',
-        });
-        const existingPrimaryNameRequest = await arioContract
-          .getPrimaryNameRequest({
-            initiator: owner.toString(),
-          })
-          .catch((e) => {
-            console.error(e);
-            return undefined;
-          });
-
-        if (
-          !existingPrimaryNameRequest ||
-          existingPrimaryNameRequest.name !== payload.name
-        ) {
-          await arioContract
-            .requestPrimaryName(
-              {
-                name: payload.name,
-                fundFrom: originalFundFrom,
-                referrer: APP_NAME,
-              },
-              WRITE_OPTIONS,
-            )
-            .catch((e) => {
-              throw new Error('Unable to request Primary name: ' + e.message);
-            });
-        }
-        // UX sleep between transactions
-        await sleep(2000);
-
-        const antProcess = ANT.init({
-          hyperbeamUrl: hyperbeamUrl,
-          signer,
-          // we're not using hyperbeam here as we're writing to the contract
-          process: new AOProcess({
-            ao: antAo,
-            processId: payload.antProcessId,
-          }),
-        });
+        result = await arioContract.setPrimaryName(
+          {
+            name: payload.name,
+          },
+          {
+            ...WRITE_OPTIONS,
+            onSigningProgress: (
+              step: keyof SetPrimaryNameProgressEvents,
+              payload: SetPrimaryNameProgressEvents[keyof SetPrimaryNameProgressEvents],
+            ) => {
+              if (step === 'requesting-primary-name') {
+                dispatch({
+                  type: 'setSigningMessage',
+                  payload: `Requesting primary name '${payload.name}'`,
+                });
+              } else if (step === 'request-already-exists') {
+                dispatch({
+                  type: 'setSigningMessage',
+                  payload: `Primary name request for '${payload.name}' already exists!`,
+                });
+              } else if (step === 'approving-request') {
+                dispatch({
+                  type: 'setSigningMessage',
+                  payload: `Approving primary name request for '${payload.name}'`,
+                });
+              }
+            },
+          },
+        );
 
         dispatch({
           type: 'setSigningMessage',
-          payload: 'Confirming Primary Name 2/2',
-        });
-        await sleep(2000);
-        result = await antProcess.approvePrimaryNameRequest({
-          name: payload.name,
-          address: owner.toString(),
-          arioProcessId: payload.arioProcessId,
+          payload: `Confirming primary name has been updated`,
         });
 
+        // wait 3 seconds and check if the primary name is set, if not show a warning saying due to cranking the primary name may take a few minutes
+        await sleep(3000);
+        await queryClient.refetchQueries({
```

Comment:
add the wallet address to the predicate, we don't need to refetch queries of other accounts we fetched.

This could happen on wallet switch.

Reply:
[447804a](https://github.com/ar-io/arns-react/pull/848/commits/447804a596d1a230a8e5550af19d9491ddcee9cd)